packages/@lwc/ssr-compiler/runtime.js

```diff
@@ -0,0 +1,137 @@
+const MULTI_SPACE = /\s+/g;
+
+class ClassList {
+    constructor(el) {
+        this.el = el;
+    }
+
+    add(...newClassNames) {
+        const className = this.el.className;
+        const set = new Set(className.split(MULTI_SPACE).filter(Boolean));
+        for (const newClassName of newClassNames) {
+            set.add(newClassName);
+        }
+        this.el.className = Array.from(set).join(' ');
+    }
+
+    contains(className) {
+        const currentClassNameStr = this.el.className;
+        return new RegExp(`\\b${className}\\b`).test(currentClassNameStr);
+    }
+
+    remove(...deprecatedClassNames) {
+        const className = this.el.className;
+        const set = new Set(className.split(MULTI_SPACE).filter(Boolean));
+        for (const newClassName of deprecatedClassNames) {
+            set.delete(newClassName);
+        }
+        this.el.className = Array.from(set).join(' ');
+    }
+
+    replace(oldClassName, newClassName) {
+        let classWasReplaced = false;
+        const className = this.el.className;
+        const listOfClasses = className.split(MULTI_SPACE).filter(Boolean);
+        listOfClasses.forEach((value, idx) => {
+            if (value === oldClassName) {
+                classWasReplaced = true;
+                listOfClasses[idx] = newClassName;
+            }
+        });
+        this.el.className = listOfClasses.join(' ');
+        return classWasReplaced;
+    }
+
+    toggle(classNameToToggle, force) {
+        const classNameStr = this.el.className;
+        const set = new Set(classNameStr.split(MULTI_SPACE).filter(Boolean));
+        if (!set.has(classNameToToggle) && force !== false) {
+            set.add(classNameToToggle);
+        } else if (set.has(classNameToToggle) && force !== true) {
+            set.remove(classNameToToggle);
+        }
+        this.el.className = Array.from(set).join(' ');
+        return set.has(classNameToToggle);
+    }
+}
+
+class LightningElement {
+    constructor(propsAvailableAtConstruction) {
+        Object.assign(this, propsAvailableAtConstruction);
+        this.isConnected = false;
+    }
+
+    __internal__setState(props, reflectedProps, attrs) {
```

Comment:
We should probably use a Symbol or WeakMap for this – otherwise people will figure out how to muck around with it.

Reply:
I opened a WI to revisit this prior to rolling out v1.